### PR TITLE
Fix whitelist behavior

### PR DESCRIPTION
**Pull request by NekoSelena**
- [x] update the configuration, so that the `enforce-whitelist` field has the correct behavior (“enforce” the whitelist on the server)
- [x] add the `whitelist` field to the configuration, allowing to activate or not the whitelist
- [x] update the functions managing the whitelist to rectify the behaviors of the modified fields of the configuration
  - [x] enforceWhitelistOnReload()
  - [x] whitelist behavior in onLoginStart (Client connexion handling)

### CODE REVIEW

**Review by Trompettesib — DISMISSED** [review]
Things in whitelist have no sens but all the code is very good !

**Comment by Trompettesib** [comment]
the fact that the object whitelist does not store the json into it is quite strange

File: cubic-server/Server.cpp [hunk 1]
```diff
@@ -208,3 +210,20 @@ void Server::_downloadFile(const std::string &url, const std::string &path)
...
+/*
+** If the server gets a /reload, players not on the whitelist
+** must be kicked from the server if enforce-whitelist is true
+** & the whitelist is in effect
+*/
+void Server::enforceWhitelistOnReload() {
+    if (_whitelist && _enforceWhitelist) {
+        for (auto &client : _clients) {
+            if (!_whitelistHandler.isPlayer(client->getPlayer()->getUuid(), client->getPlayer()->getUsername(), _whitelistData).first) {
+                client->stop("You are not whitelisted on this server.");
```
[the remark this answers is not in the document]
Comment (STMiki) [reply cubic-server/Server.cpp:223]: It seems I was drunk when I created the stop functions, don't use them, most of them are useless.

Will fix later :sweat:

File: cubic-server/Server.hpp [hunk 2]
```diff
@@ -45,10 +46,20 @@ class Server
 
     void stop();
 
+    void enforceWhitelistOnReload();
+
     const Configuration::ConfigHandler &getConfig() const {
         return _config;
     }
 
+    WhitelistHandling::Whitelist getWhitelist() const {
```
Comment (Trompettesib) [inline cubic-server/Server.hpp:55]: This is strange that we can't use a const ref here but it's ok

**Review by Trompettesib — APPROVED** [review]
👌

File: cubic-server/Server.hpp (hunk 2 above)
Comment (NekoSelena) [reply cubic-server/Server.hpp:55]: You know it ends up crashing if we make it const.

Comment (STMiki) [reply cubic-server/Server.hpp:55]: It's because `isPlayerWhitelisted` is not const

Comment (NekoSelena) [reply cubic-server/Server.hpp:55]: Yes, because it can't be because of the iterator, it crashes if it is const

Comment (STMiki) [reply cubic-server/Server.hpp:55]: And thats why you use auto on iterator.

Try with `const_iterator` 😉

Comment (STMiki) [reply cubic-server/Server.hpp:55]: Or just use auto

Comment (NekoSelena) [reply cubic-server/Server.hpp]: Fixed 👍

**Review by Trompettesib — APPROVED** [review]
👌

**Review by STMiki — APPROVED** [review]

**Review by huntears — APPROVED** [review]

**Review by Trompettesib — APPROVED** [review]

**Review by STMiki — APPROVED** [review]

**Pull request merged**